### DISCUSSION TITLE
CLAP_NOTE_EXPRESSION_VOLUME

### DISCUSSION THREAD

**Discussion by sagantech** [free-audio/clap#172] [Q&A] [answered]
So I have been living under a rock for a while but now that I am out I have discovered note expression.

Anyway I ran into this comment in events.h:

> // with 0 < x <= 4, plain = 20 * log(x)
>    CLAP_NOTE_EXPRESSION_VOLUME,
> 
What does that mean? If a host would like a plug-in to set the volume to 0 db, what should 'value' be set to? Also is CLAP_NOTE_EXPRESSION_VOLUME required to be handled by a plug-in? Is there a way of telling which note_expressions are handled by a given plug-in?

**Comment by baconpaul** [comment, marked as answer]
This is basically gain so 1 is 0db. 0 is -inf db.
If a plugin note ports advertises clap note syntax it will at least receive these events.
Both surge and the clap saw demo respond to volume pitch and pan and surge also routes pressure and brightness

**Reply by sagantech** [reply]
Thank you. I had presumed that but a recent CLAP plugin (Apricot) was responding to pan and not to gain so I figured I must be doing it wrong.
